app/ratelimit_test.go

```diff
@@ -27,15 +27,15 @@ func genRateLimitSettings(useAuth, useIP bool, header string) *model.RateLimitSe
 
 func TestNewRateLimiterSuccess(t *testing.T) {
 	settings := genRateLimitSettings(false, false, "")
-	rateLimiter, err := NewRateLimiter(settings)
+	rateLimiter, err := NewRateLimiter(settings, nil)
```

Comment:
Need a test for when this value is `non-nil`.

Reply:
I've added some unit tests to cover all cases, also some on the key generation itself.